src/utils.js

```diff
@@ -64,6 +64,83 @@ CKEDITOR.define( function() {
 		 */
 		isObject: function( obj ) {
 			return typeof obj === 'object' && !!obj;
-		}
+		},
+
+		/**
+		 * A mixin function to be used to implement the static `extend()` method in classes. It allows for easy creation
+		 * of subclasses.
+		 *
+		 * @param {Object} [proto] Extensions to be added to the subclass prototype.
+		 * @param {Object} [statics] Additional static properties to be added to the subclass constructor.
+		 * @returns {Object} When executed as a static method of a class, it returns the new subclass constructor.
+		 */
+		extendMixin: function( proto, statics ) {
+			var that = this;
+			var child = ( proto && proto.hasOwnProperty( 'constructor' ) ) ?
+					proto.constructor :
+					function() {
+						that.apply( this, arguments );
+					};
+
+			// Copy the statics.
+			utils.extend( child, this, statics );
+
+			// Use the same prototype.
+			child.prototype = Object.create( this.prototype );
+
+			// Add the new prototype stuff.
+			if ( proto ) {
+				utils.extend( child.prototype, proto );
+			}
+
+			return child;
+		},
+
+		/**
+		 * Creates a spy function (ala Sinon.js) that can be used to inspect call to it.
+		 *
+		 * The following are the present features:
+		 *
+		 *  * spy.called: property set to `true` if the function has been called at least once.
+		 *
+		 * @returns {Function} The spy function.
+		 */
+		spy: function() {
+			var spy = function() {
+				spy.called = true;
+			};
+
+			return spy;
+		},
+
+		/**
+		 * Returns a unique id. This id is a number (starting from 1) which will never get repeated on successive calls
+		 * to this method.
+		 *
+		 * @returns {Number} A number representing the id.
+		 */
+		uid: ( function() {
+			var next = 1;
+
+			return function() {
+				return next++;
+			};
+		} )()
 	};
+
+	return utils;
 } );
+
+/**
```

Comment:
That's an interesting idea. I'm all for it, maybe named `BasicClass` or `Essentials` to avoid a name so close to the `class` keyword.

It assumes that classes will always have `extend()`, be `emitters`, etc... but I think it is an acceptable assumption.